### ISSUE TITLE
Should we prefer `|` to `+` when combining bit flags?

### ISSUE THREAD

**Issue by Lingxi-Li** [actor-framework/actor-framework#450]
In fact, I have never seen `+` be used to combine bit flags before. CAF seems to have adopted this convention (e.g., to combine actor `spawn_options`). In the beginning, I have thought that this was really a smart idea for it's more intuitive than `|`. The addition operator `+` conveys a sense of combining, while `|` practically conveys nothing unless you are familiar with bit-wise operations. For this reason, I adopted it as my own convention too.

But then, I see the following (quoted from *Programming C# 5.0* by Ian Griffiths):
> When combining flag-based enumeration values, we normally use the bitwise OR operator... You might come across code that uses `+` instead of `|`. This works for some combinations, but `Ingredients.TheFullEnglish + Ingredients.Eggs` would be a value of `0x80`, which doesn’t correspond to anything, so it’s safer to stick with `|`.

Basically, it says that `|` is [idempotent](https://en.wikipedia.org/wiki/Idempotence) while `+` is not. For example, `monitored | monitored` is still `monitored`, while `monitored + monitored` is not. So, maybe, it is not a good idea to use `+` instead of `|`?

**Comment by Neverlord** [comment]
Oh, you should definitely use `|` for combining bit flags! I'm actually surprised anyone would even consider using `+` for this purpose.

However, `spawn_option` is an enum class. This means the compiler isn't allowed to do its "silently convert to `int` and perform bit operations and whatever on it" shenanigans.

I used `+` instead of `|` (or even `&`) to avoid confusion with bitmask syntax...

Oh, and combining spawn options is idempotent by the way:

```cpp
assert(monitored + monitored == monitored); // compiles + runs
assert(monitored | monitored == monitored); // does not compile
```

Granted, the `+` operator is not the best fit. But C++ has no union operator.

**Comment by Lingxi-Li** [comment]
Oh I see. My bad. Should have read the code more carefully. Thanks for the clarification. The code is good :)
